classes/class.ilBigBlueButtonProtocol.php

```diff
@@ -221,6 +223,37 @@ private function isPDFValid(string $pdf){
         
         return filter_var($pdf, FILTER_VALIDATE_URL) ? true : false;
     }
+
+    
+    public function getMaximumSessionsAvailable($meeting_id = null)
+    {
+        $participants_count = 0;
+        $sessions_available = array(
+            "current_meeting_userlimit" => false,
+            "max_sessions" => false,
+        );
+        $available = array();
+        foreach($this->meetings->getMeetings() as $meeting){
+            $participants_count = $participants_count + $meeting->getParticipantCount();
+            $userlimit_exceeded = ($meeting->getMaxUsers() - $meeting->getParticipantCount() -1 <= 0);
```

Comment:
I'm not sure. But i think if you want to check the  maxUsers, you must set while create the meeting the maximum allowed Participants (setMaxParticipants($maxParticipants)). But be aware, BBB then forbids you to enter with other users, including moderators.

Reply:
Dear @xJuvi ,
yeah the maxUsers returns the value set by setMaxParticipants during the creation of the meeting. do you mean that setting setMaxParticipants always forbids users+moderators to enter the room?

What was happening before was that once you set a maximum number of participants and the number of users entering the room exceed this maxParticipants, then the new users entering the room would not be allowed to join the room and there was no mechanism to inform the user or the moderator that the maxParticipants has been exceeded. checking maxUsers here is a way to make sure that the set number is not exceeded. If it is there will be error message notifiying the user and the moderator about the issue.

Reply:
Hi @jeph864 
sorry, my answer was unclear. So if the maximum user isn't exceeded, then everybody can join. I think (but it's untested), if the maximum is exceeded, nobody, no users and not moderators can join.

Your logic is correct and your code works. But with this plugin we never set the maximum allowed participants. We need to add this options, if activated, on meeting creation. Otherwise the "getMaxUser()" value used the BBB default value and the calculation of free seats can be wrong because the BBB server can have other limits as configured in this plugin.

That's my only suggestion for this PR. Should i add this to your PR oder would you do it by yourself?

Kind regards

Reply:
@xJuvi , Oh I understand. I will add the following check: 
```suggestion
            $userlimit_exceeded =( $this->object->getMaxParticipants() > 0 && ($meeting->getMaxUsers() - $meeting->getParticipantCount() -1 <= 0));
```

Reply:
Yes, so i should calculate correctly. Thanks for you fast response, @jeph864